db/cache/charts.go

```diff
@@ -913,82 +931,93 @@ func blockTimes(blocks ChartUints) (ChartUints, ChartUints) {
+		return ChartUints{}
 	}
-	times := make(ChartUints, 0, len(ticks)-1)
-	avgs := make(ChartUints, 0, len(ticks)-1)
-	workingOn := ticks[0]
+	avgDiffs := make(ChartUints, 0, len(ticks)-1)
 	nextIdx := 1
 	next := ticks[nextIdx]
 	lastIdx := 0
 	for i, t := range blocks {
 		if t > next {
 			_, pts := blockTimes(blocks[lastIdx:i])
-			avgs = append(avgs, pts.Avg(0, len(pts)))
-			times = append(times, workingOn)
+			avgDiffs = append(avgDiffs, pts.Avg(0, len(pts)))
 			nextIdx++
 			if nextIdx > len(ticks)-1 {
 				break
 			}
-			workingOn = next
 			lastIdx = i
 			next = ticks[nextIdx]
 		}
 	}
-	return times, avgs
+	return avgDiffs
 }
 
-func blockSizeChart(charts *ChartData, zoom ZoomLevel) ([]byte, error) {
-	switch zoom {
-	case BlockZoom:
+func blockSizeChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, error) {
+	if bin == DayBin && axis == HeightAxis {
+		return charts.encode(charts.Days.Height, charts.Days.BlockSize)
+	}
+	switch bin {
+	case BlockBin:
 		return charts.encode(charts.Blocks.Time, charts.Blocks.BlockSize)
-	case DayZoom:
+	case DayBin:
 		return charts.encode(charts.Days.Time, charts.Days.BlockSize)
 	}
-	return nil, InvalidZoomErr
+	return nil, InvalidBinErr
 }
 
-func blockchainSizeChart(charts *ChartData, zoom ZoomLevel) ([]byte, error) {
-	switch zoom {
-	case BlockZoom:
+func blockchainSizeChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, error) {
+	if bin == DayBin && axis == HeightAxis {
+		return charts.encode(charts.Days.Height, accumulate(charts.Days.BlockSize))
+	}
+	switch bin {
+	case BlockBin:
 		return charts.encode(charts.Blocks.Time, accumulate(charts.Blocks.BlockSize))
-	case DayZoom:
+	case DayBin:
 		return charts.encode(charts.Days.Time, accumulate(charts.Days.BlockSize))
 	}
-	return nil, InvalidZoomErr
+	return nil, InvalidBinErr
 }
 
-func chainWorkChart(charts *ChartData, zoom ZoomLevel) ([]byte, error) {
-	switch zoom {
-	case BlockZoom:
+func chainWorkChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, error) {
+	if bin == DayBin && axis == HeightAxis {
+		return charts.encode(charts.Days.Height, charts.Days.Chainwork)
+	}
+	switch bin {
+	case BlockBin:
 		return charts.encode(charts.Blocks.Time, charts.Blocks.Chainwork)
-	case DayZoom:
+	case DayBin:
 		return charts.encode(charts.Days.Time, charts.Days.Chainwork)
 	}
-	return nil, InvalidZoomErr
+	return nil, InvalidBinErr
 }
 
-func coinSupplyChart(charts *ChartData, zoom ZoomLevel) ([]byte, error) {
-	switch zoom {
-	case BlockZoom:
+func coinSupplyChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, error) {
+	if bin == DayBin && axis == HeightAxis {
+		return charts.encode(charts.Days.Height, accumulate(charts.Days.NewAtoms), charts.Days.Height)
+	}
+	switch bin {
+	case BlockBin:
 		return charts.encode(charts.Blocks.Time, accumulate(charts.Blocks.NewAtoms))
-	case DayZoom:
+	case DayBin:
 		return charts.encode(charts.Days.Time, accumulate(charts.Days.NewAtoms), charts.Days.Height)
```

Comment:
I meant about this when I was referencing the previous implementation.

Reply:
Does it have any significance?

Reply:
Yes. The first argument to `encode` is `charts.Days.Time` here. You have changed it to `charts.Days.Height`. Before, it was a time axis, a coin supply axis, and a height axis. Now it is two identical axes of height and one axis of coin supply.